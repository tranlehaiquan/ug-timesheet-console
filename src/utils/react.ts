import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
