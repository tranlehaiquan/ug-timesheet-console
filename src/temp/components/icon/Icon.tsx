import * as React from "react";
import icons from "./iconPaths";
import classNames from "classnames";

export type IconNames = keyof typeof icons;

interface IProps extends React.SVGAttributes<SVGElement> {
  /**
   * Name of the icon
   */
  name: IconNames;
  /**
   * className applied to the containing div
   */
  className?: string;
  /**
   * Size of the icon to override the default (pixels)
   */
  size?: number;
  /**
   * Function to call on click
   */
  onClick?: (event?: React.MouseEvent<SVGElement>) => void;
}

const DEFAULT_SIZE = 18;

/**
 * The Icon component should be used whenever an icon needs to be displayed. It will naturally inherit any text colour but this can be changed by supplied a class directly. Default size of 18px/18px.
 */
export const Icon = ({
  name,
  className,
  size,
  onClick,
  ...otherProps
}: IProps) => {
  const Component = icons[name];
  return (
    <Component
      {...otherProps}
      height={size || DEFAULT_SIZE}
      width={size || DEFAULT_SIZE}
      onClick={onClick}
      className={classNames("sk-flex-shrink-0 sk-fill-current", className)}
    />
  );
};
