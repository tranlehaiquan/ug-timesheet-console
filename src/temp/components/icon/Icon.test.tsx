import * as React from "react";
import { Icon } from "./Icon";
import { render, shallow } from "enzyme";

describe("Icon", () => {
  test("renders correctly", () => {
    // Arrange
    const arrowIcon = render(<Icon name="arrowLeft" />);

    // Assert
    expect(arrowIcon).toMatchSnapshot();
  });

  test("renders with additional classes", () => {
    // Arrange
    const jobsIcon = render(<Icon name="jobs" className="extra-class" />);

    // Assert
    expect(jobsIcon).toMatchSnapshot();
  });

  test("changes when you change icon name", () => {
    // Arrange
    const arrowIcon = render(<Icon name="arrowLeft" />);
    const chatIcon = render(<Icon name="chat" />);

    // Assert
    expect(arrowIcon).not.toEqual(chatIcon);
  });

  test("calls onclick prop if supplied", () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(<Icon name="chat" onClick={onClickSpy} />);
    wrapper.simulate("click");
    expect(onClickSpy).toBeCalled();
  });

  test("it passes through other props", () => {
    const wrapper = shallow(<Icon name="chevronDown" data-sk-name="test" />);
    expect(wrapper.find('[data-sk-name="test"]').length).toBe(1);
  });
});
