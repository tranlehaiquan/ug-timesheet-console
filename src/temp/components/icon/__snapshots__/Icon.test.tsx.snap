// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<svg
  class="sk-flex-no-shrink sk-fill-current"
  height="18"
  id="Layer_1"
  viewBox="0 0 18 18"
  width="18"
>
  <path
    d="M4.4 10l3.3 3.3c.4.4.4 1 0 1.4-.4.4-1 .4-1.4 0l-5.1-5c-.4-.4-.4-1 0-1.4l5-5.1c.3-.1.5-.2.8-.2.3 0 .5.1.7.3.4.4.4 1 0 1.4L4.4 8H16c.6 0 1 .4 1 1s-.4 1-1 1H4.4z"
  />
</svg>
`;

exports[`Icon renders with additional classes 1`] = `
<svg
  class="sk-flex-no-shrink sk-fill-current extra-class"
  height="18"
  viewBox="0 0 18 18"
  width="18"
>
  <path
    d="M17 6H1c-.6 0-1 .4-1 1v10c0 .6.4 1 1 1h16c.6 0 1-.4 1-1V7c0-.6-.4-1-1-1zM2 16V8h14v8H2zM11.2 1H6.8C5.8 1 5 1.8 5 3v1c0 .6.4 1 1 1s1-.4 1-1V3h4v1c0 .6.4 1 1 1s1-.4 1-1V3c0-1.2-.8-2-1.8-2z"
  />
</svg>
`;
